src/components/dialog/dialog.pw.tsx

```diff
@@ -0,0 +1,596 @@
+
+    await expect(dialog).toBeVisible();
+  });
+
+  test("when a function is passed to the onCancel prop, clicking close icon should call that function", async ({
+    mount,
+    page,
+  }) => {
+    let called = false;
+    await mount(
+      <DialogComponent
+        onCancel={() => {
+          called = true;
+        }}
+      />
+    );
+
+    await page.getByLabel("Close").click();
+    expect(called).toBeTruthy();
+  });
+
+  test("when clicking outside the Dialog, the Dialog should not close", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<DialogComponent />);
+    await page.mouse.click(0, 0);
+    await expect(page.getByRole("dialog")).toBeVisible();
+  });
+
+  test("when aria-label prop is passed, it should be used as the aria label for Dialog", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<DialogComponent aria-label="aria label for dialog" />);
+    await expect(page.getByRole("dialog")).toHaveAttribute(
+      "aria-label",
+      "aria label for dialog"
+    );
+  });
+
+  test("when aria-describedby prop is passed, the aria-describedby attribute of the Dialog should be set to the prop's value", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<DialogComponent aria-describedby="aria description" />);
+    await expect(page.getByRole("dialog")).toHaveAttribute(
+      "aria-describedby",
+      "aria description"
+    );
+  });
+
+  test("when aria-labelledby prop is passed and title prop is not, the aria-labelledby attribute of the Dialog should be set to the prop's value", async ({
+    mount,
+    page,
+  }) => {
+    await mount(
+      <DialogComponent title={undefined} aria-labelledby="label-element-id" />
+    );
+    await expect(page.getByRole("dialog")).toHaveAttribute(
+      "aria-labelledby",
+      "label-element-id"
+    );
+  });
+
+  test("when disableClose prop is passed, close icon should be disabled", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<DialogComponent disableClose />);
+    await expect(page.getByLabel("Close")).toBeDisabled();
+  });
+
+  test("when help prop is provided, hovering over the rendered help icon displays help text", async ({
+    mount,
+    page,
+  }) => {
+    await mount(
+      <DialogComponent title="Sample Dialog" help="Some help text" />
+    );
+
+    const helpIcon = page.getByLabel("help");
+    await helpIcon.hover();
+
+    await expect(page.getByRole("tooltip")).toHaveText("Some help text");
+  });
+
+  test("when Dialog is opened, the first focusable element should be focused", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<DialogWithFirstFocusableElement />);
+    await expect(
+      page.getByRole("button").filter({ hasText: "Press me" })
+    ).toBeFocused();
+  });
+
+  test("when disableAutoFocus prop is passed, the first focusable element should not be focused", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<DialogWithFirstFocusableElement disableAutoFocus />);
```

Comment:
suggestion: (non-blocking) it would be better to add a space to separate the mount from the assertion to follow the same pattern as with the other test cases.